pkg/client/imagestreams.go

```diff
@@ -108,3 +115,30 @@ func (c *imageStreams) UpdateStatus(stream *imageapi.ImageStream) (result *image
 	err = c.r.Put().Namespace(c.ns).Resource("imageStreams").Name(stream.Name).SubResource("status").Body(stream).Do().Into(result)
 	return
 }
+
+// Import makes a call to the server to retrieve information about the requested images or to perform an import. ImageStreamImport
+// will be returned if no actual import was requested (the to fields were not set), or an ImageStream if import was requested.
+func (c *imageStreams) Import(isi *imageapi.ImageStreamImport) (*imageapi.ImageStreamImport, error) {
+	result := &imageapi.ImageStreamImport{}
+	if err := c.r.Post().Namespace(c.ns).Resource("imageStreamImports").Body(isi).Do().Into(result); err != nil {
+		return nil, transformUnsupported(err)
```

Comment:
I'm not sure I like intercepting the error and preventing our client API from returning the raw error that is usually an `APIStatus`.

Also, wouldn't we expect this request to be gated by a capability check based on the discovery doc?